ddtrace/tracer/stats.go

```diff
@@ -104,18 +104,9 @@ func (c *concentrator) runFlusher(tick <-chan time.Time) {
 	for {
 		select {
 		case now := <-tick:
-			p := c.flush(now)
-			if len(p.Stats) == 0 {
-				// nothing to flush
-				continue
-			}
-			c.statsd().Incr("datadog.tracer.stats.flush_payloads", nil, 1)
-			c.statsd().Incr("datadog.tracer.stats.flush_buckets", nil, float64(len(p.Stats)))
-			if err := c.cfg.transport.sendStats(&p); err != nil {
-				c.statsd().Incr("datadog.tracer.stats.flush_errors", nil, 1)
-				log.Error("Error sending stats payload: %v", err)
-			}
+			c.flush(now, withoutCurrentBucket)
 		case <-c.stop:
+			c.flush(time.Now(), withCurrentBucket)
```

Comment:
This call seems to cause a deadlock with the `ingester` test case when `c.Stop()` is called 🤔